<?xml version="1.0"?>
<launch>

  <arg name="serial_no"       default=""/>
  <arg name="device_type"     default="d415"/>
  <arg name="json_file_path"  default=""/>

  <arg name="camera"  default="camera"/>
  <arg name="manager"  default="realsense2_camera_manager"/>
  <arg name="rgb"  default="color"/>
  <arg name="depth_registered"  default="depth_registered"/>

  <include file="$(find realsense2_camera)/launch/rs_rgbd.launch">
    <arg name="camera"         value="$(arg camera)"/>
    <arg name="manager"         value="$(arg manager)"/>
    <arg name="rgb"         value="$(arg rgb)"/>
    <arg name="serial_no"         value="$(arg serial_no)"/>
    <arg name="device_type"       value="$(arg device_type)"/>
    <arg name="json_file_path"    value="$(arg json_file_path)"/>
    <arg name="enable_depth"      value="true"/>
    <arg name="enable_color"      value="true"/>
    <arg name="enable_infra1"     value="false"/>
    <arg name="enable_infra2"     value="false"/>
    <arg name="enable_fisheye"    value="false"/>
    <!-- <arg name="enable_imu"        value="false"/> -->
    <arg name="color_fps" value="15" />
    <arg name="depth_fps" value="15" />
  </include>

  <group ns="$(arg camera)">
    <node pkg="nodelet" type="nodelet" name="shm_image_saver"
      args="load shm_image_save/plugin_nodelet_shm_saver $(arg manager)" >
      <remap from="camera_info"		to="$(arg rgb)/camera_info" />
      <remap from="camera_image"		to="$(arg rgb)/image_raw" />
<!--
      <remap from="depth_registered/points"		to="$(arg depth_registered)/points" />
-->
    </node>
  </group>
</launch>
